src/core/Akka/IO/TcpConnection.cs

```diff
@@ -619,31 +619,84 @@ protected void AcquireSocketAsyncEventArgs()
             if (ReceiveArgs != null) throw new InvalidOperationException($"Cannot acquire receive SocketAsyncEventArgs. It's already has been initialized");
             if (SendArgs != null) throw new InvalidOperationException($"Cannot acquire send SocketAsyncEventArgs. It's already has been initialized");
 
-            ReceiveArgs = Tcp.SocketEventArgsPool.Acquire(Self);
+            ReceiveArgs = CreateSocketEventArgs(Self);
             var buffer = Tcp.BufferPool.Rent();
             ReceiveArgs.SetBuffer(buffer.Array, buffer.Offset, buffer.Count);
 
-            SendArgs = Tcp.SocketEventArgsPool.Acquire(Self);
+            SendArgs = CreateSocketEventArgs(Self);
         }
 
         private void ReleaseSocketAsyncEventArgs()
         {
             if (ReceiveArgs != null)
             {
                 var buffer = new ByteBuffer(ReceiveArgs.Buffer, ReceiveArgs.Offset, ReceiveArgs.Count);
-                Tcp.SocketEventArgsPool.Release(ReceiveArgs);
+                ReleaseSocketEventArgs(ReceiveArgs);
                 //TODO: there is potential risk, that socket was working while released. In that case releasing buffer may not be safe.
                 Tcp.BufferPool.Release(buffer);
```

Comment:
Is this line still necessary?

Reply:
This depends on `Tcp.BufferPool` implementation used. If we will remove pooling completely and will allocate new byte buffers each time when new connection is opened (that's what `DisabledBufferPool` is doing) - that we will be able to remove this line. 

For now just updated to reference DirectBufferPool class

Reply:
cc @Aaronontheweb